### PR TITLE
Add another util function

### PR DESCRIPTION
**Pull request by CGNonofr**

### CODE REVIEW

File: src/tools.ts [hunk 1]
```diff
@@ -269,3 +269,58 @@ export async function collapseCodeSections (editor: monaco.editor.IStandaloneCod
...
+
+export function mapClipboard (
+  editor: monaco.editor.IStandaloneCodeEditor, {
+    toClipboard = v => v,
+    fromClipboard = v => v
+  }: {
+    toClipboard: (v: string) => string
+    fromClipboard: (v: string) => string
+  }): void {
+  let copiedText = ''
```
Comment (samuel-olivier) [inline src/tools.ts:281]: It's not super clear why we have to maintain a `copiedText`.

File: src/tools.ts [hunk 2]
```diff
@@ -269,3 +269,58 @@ export async function collapseCodeSections (editor: monaco.editor.IStandaloneCod
...
+  editor.addCommand(monaco.KeyMod.CtrlCmd | monaco.KeyCode.KeyC, () => {
```
Comment (samuel-olivier) [inline src/tools.ts:282]: Not sure why it's necessary, wouldn't it call `editor.action.clipboardCopyAction` by default ?

**Review by samuel-olivier — CHANGES_REQUESTED** [review]
Some comments would help I think

File: src/tools.ts (hunk 1 above)
Comment (CGNonofr) [reply src/tools.ts]: I simplified the function, it still works on chrome and firefox but I suspect this code was to make it work on edge

File: src/tools.ts (hunk 2 above)
Comment (CGNonofr) [reply src/tools.ts]: It's not even required

File: src/tools.ts [hunk 3]
```diff
@@ -269,3 +270,52 @@ export async function collapseCodeSections (editor: monaco.editor.IStandaloneCod
...
+      return
+    }
+    const content = clipdata.getData('Text')
+    const transformed = toClipboard(content)
+    if (transformed !== content) {
+      // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+      if (clipdata.types != null) {
+        clipdata.types.forEach((type) => clipdata.setData(type, toClipboard(content)))
+      } else {
+        clipdata.setData('text/plain', toClipboard(content))
```
Comment (samuel-olivier) [inline src/tools.ts:308]: I guess we can reuse `transformed` instead of `toClipboard(content)`

File: src/tools.ts (hunk 1 above)
Comment (samuel-olivier) [reply src/tools.ts]: Even now that Edge is Chromium based ?

Comment (CGNonofr) [reply src/tools.ts]: Probably not... I don't know and I don't have the possibility to check, nobody has windows there

**Pull request closed without merging**